src/Authentication/Authentication/Helpers/EncodeUriPath.cs

```diff
@@ -15,6 +15,8 @@ public static class EncodeUriPaths
     {
         public static Uri EscapeDataStrings(this Uri uri)
         {
+            if(uri.Query.Length > 0)
```

Comment:
The only question I have is what happens when the Uri has both path segments and query parameters?
e.g. `https://graph.microsoft.com/v1.0/users/userswith#character?$select=id,displayName`

Reply:
Nice catch!. I've made corrections to address this.